### DISCUSSION TITLE
TypeError: Do not know how to serialize a BigInt

### DISCUSSION THREAD

**Discussion by zhangxiaoshang** [wevm/wagmi#2198] [Q&A]
I used wagmi/cli to generate hooks for an ERC20 contract and used the generated usePrepareErc20Approve and useErc20Approve functions to perform an approve. However, the program threw a TypeError: Do not know how to serialize a BigInt exception, and I am unsure how to resolve this issue.

Furthermore, I would like to mention that while using the usePrepareErc20Approve function, I used a string value for the second argument of args and asserted it as any, which allowed the program to run. However, the args generated by wagmi/cli had a second parameter of type BigInt, and I would like to use this type instead of asserting it as any.
 
 some code:
```tsx
  const { config } = usePrepareErc20Approve({
    address,
    chainId: bscTestnet.id,
    args: [spender, utils.parseEther("1").toBigInt()]
  });
  const { write, status: mintStatus } = useErc20Approve({
    ...config
  });
```

error:
![image](https://user-images.githubusercontent.com/19658966/232959249-4cc03ffd-bdfa-4a8b-8678-8851d2df749d.png)

package.json
```json
"dependencies": {
    "@emotion/react": "^11.10.6",
    "@emotion/styled": "^11.10.6",
    "@fontsource/roboto": "^4.5.8",
    "@mui/icons-material": "^5.11.16",
    "@mui/material": "^5.12.1",
    "@types/node": "18.15.11",
    "@types/react": "18.0.37",
    "@types/react-dom": "18.0.11",
    "@web3modal/ethereum": "^2.3.0",
    "@web3modal/react": "^2.3.0",
    "eslint": "8.38.0",
    "eslint-config-next": "13.3.0",
    "ethers": "5.6.9",
    "next": "13.3.0",
    "notistack": "^3.0.1",
    "nprogress": "^0.2.0",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "sass": "^1.62.0",
    "typescript": "5.0.4",
    "wagmi": "^0.12.10"
  },
  "devDependencies": {
    "@wagmi/cli": "^0.1.15"
  }
```

and wagmi.config.ts
```ts
import { defineConfig } from "@wagmi/cli";
import { etherscan, react } from "@wagmi/cli/plugins";
import { erc20ABI } from "wagmi";

export default defineConfig({
  out: "src/generated.ts",
  contracts: [
    {
      name: "erc20",
      abi: erc20ABI,
    }
  ],
  plugins: [
    react(),
  ],
});

```

**Comment by tmm** [comment]
Can you drop `.toBigInt()`? ethers should convert this internally.

**Reply by zhangxiaoshang** [reply]
The reason for using the bitInt type is to comply with TypeScript type checking, as the hook generated by wagmi/cli requires the second parameter to be of type bigint.
